CMakeLists.txt

```diff
@@ -615,6 +616,12 @@ function(
         set(enable_long_double_test true)
     endif()
 
+    if(build_type MATCHES "release")
+        # Set additional performance options
+        set(newlib_nano_malloc "false")
+    else()
```

Comment:
IIUC buildtype is a meson option https://mesonbuild.com/Builtin-options.html which can be one of plain, debug, debugoptimized, release, minsize, custom.

Would it be best to check for minsize to set newlib_nano_malloc to true, and everything else is the larger version.